### PR TITLE
Added a simple implementation of the account service (Task to review PR)

### PR DESCRIPTION
**Pull request by petar-haydutov**
Added a simple implementation of the account service (Task to review PR)

### CODE REVIEW

File: src/main/java/com/yotpo/service/AccountServiceImpl.java [hunk 1]
```diff
@@ -14,17 +14,20 @@ public class AccountServiceImpl implements AccountService {
...
     public void createAccount(Account account) {
-        //do nothing for now
+        accounts.add(account);
     }
 
     @Override
     public Account getAccount(long id) {
-        return null;
+        return accounts.stream()
+                .filter(account -> account.getAccountKey() == AccountKey.valueOf(id))
```
Comment (petar-haydutov) [inline src/main/java/com/yotpo/service/AccountServiceImpl.java:28]: The current implementation uses `ArrayList` for account storage, resulting in O(n) lookup time in `getAccount()`. This will become a performance bottleneck down the line. I'd recommend we switch to something like a `ConcurrentHashMap<AccountKey, Account>` instead. This will provide us with O(1) lookups, thread safety without locking the whole map in case this is called from multiple places at once and there will be minimal memory overhead.

Line 28 uses `==` to compare AccountKey objects, which compares references rather than values. This will cause lookups to fail. AccountKey will need `equals()` and `hashCode()` implementations to work correctly as a HashMap key.

Let's also add some unit tests covering things like account creation and retrieval, duplicate account handling and lookup of non-existent accounts

I'm happy to discuss these suggestions further or pair on the implementation if it would be helpful
